tools/add_triage_signature.py

```diff
@@ -745,9 +728,14 @@ def add_signatures(jclient, url, issue_key, should_update=False, signatures=None
         should_update = True
         signatures_to_add = [v for k, v in name_to_signature.items() if k in signatures]
 
+    # boltons LRU is practically a size-limited defaultdict where the least recently used keys get evicted automatically
```

Comment:
I guess I could replace the dict with a function that receives the key as input and the returns the object. Then decorate it with `lru_cache`. It'll "save" me from having to pass the log cache to all the signatures because it kinda turns the dict into a global variable that masquerades as a function.

Will do that